core/templates/pages/contributor-dashboard-page/contributor-stats/contributor-stats.component.ts

```diff
@@ -0,0 +1,394 @@
+
+    const userContributionRights =
+      await this.userService.getUserContributionRightsDataAsync();
+
+    if (userContributionRights === null) {
+      throw new Error('Cannot fetch user contribution rights.');
+    }
+    const reviewableLanguageCodes = (
+      userContributionRights.can_review_translation_for_language_codes);
+    this.userCanReviewTranslationSuggestions = (
+      reviewableLanguageCodes.length > 0);
+    this.userCanReviewQuestionSuggestions = (
+      userContributionRights.can_review_questions);
+    this.userCanSuggestQuestions = (
+      userContributionRights.can_suggest_questions);
+
+    if (this.userCanReviewTranslationSuggestions) {
+      this.options.push(this.translationReviewOption);
+    }
+    if (this.userCanSuggestQuestions) {
+      this.options.push(this.questionContributionOption);
+    }
+    if (this.userCanReviewQuestionSuggestions) {
+      this.options.push(this.questionReviewOption);
+    }
+
+    await this.fetchStats();
+  }
+
+  toggleDropdown(): void {
+    this.dropdownShown = !this.dropdownShown;
+  }
+
+  toggleMobileDropdown(): void {
+    this.mobileDropdownShown = !this.mobileDropdownShown;
+  }
+
+  async selectOption(contributionType: string): Promise<void> {
+    this.type = contributionType;
+    const currentOption = this.options.find(
+      (option) => option.contributionType === contributionType);
+    this.selectedContributionType = currentOption?.displayName;
+    this.dropdownShown = false;
+    this.mobileDropdownShown = false;
+  }
+
+  async fetchStats(): Promise<void> {
+    const response = await this.contributionAndReviewStatsService.fetchAllStats(
+      this.username);
+
+    if (response.translation_contribution_stats.length > 0) {
+      response.translation_contribution_stats.map((stat) => {
+        const language = this.languageUtilService.getAudioLanguageDescription(
+          stat.language_code);
+        if (
+          this.statsData.translationContribution === null ||
+          this.statsData.translationContribution === undefined) {
+          throw new Error('Translation contributions are undefined.');
+        }
+        if (!this.statsData.translationContribution[language]) {
+          this.statsData.translationContribution[language] = new PageableStats(
+            [this.createTranslationContributionStat(stat)]);
+        } else {
+          this.statsData?.translationContribution[language]?.data?.push(
+            this.createTranslationContributionStat(stat));
+        }
+      });
+    }
+
+    if (response.translation_review_stats.length > 0) {
+      response.translation_review_stats.map((stat) => {
+        const language = this.languageUtilService.getAudioLanguageDescription(
+          stat.language_code);
+        if (
+          this.statsData.translationReview === null ||
+          this.statsData.translationReview === undefined) {
+          throw new Error('Translation reviews are undefined.');
+        }
```

Comment:
I think we can actually remove this check if we instantiate the `this.statsData.translationReview` on declaration above.

Reply:
Done